ChangeTests/ChangeUnitTests.cs

```diff
@@ -0,0 +1,96 @@
+namespace ChangeTests
+{
+    [TestClass]
+    public class ChangeUnitTests
+    {
+        private Dictionary<int, int> availableCoins;
+        MyCash myCash;
```

Comment:
I would propose you to not share the instance of MyCash among tests but create it for each tests in isolation. Thus, you won't use these shared variables. If we want to run these unit tests in parallel they will fail. This kind of tests should be independent of each other as much as possible.
You can also use a chance to use different available coins in different test cases. Please use the minimally needed coins to check the case.

Reply:
done